aws-android-sdk-cognitoauth/src/main/java/com/amazonaws/mobileconnectors/cognitoauth/AuthClient.java

```diff
@@ -271,18 +302,42 @@ public void signOut(final boolean clearLocalTokensOnly) {
     /**
      * Signs-out a user.
      * <p>
-     *     Clears cached tokens for the user. Launches the sign-out Cognito web end-point to
-     *     clear all Cognito Auth cookies stored by Chrome.
+     *     Launches the sign-out Cognito web end-point to clear all Cognito Auth cookies stored
+     *     by Chrome. Cached tokens will be deleted if sign-out redirect is completed.
      * </p>
      *
      * @param clearLocalTokensOnly true if signs out the user from the client,
      *                             but the session may still be alive from the browser.
      * @param browserPackage String specifying the browser package to launch the specified url.
      */
     public void signOut(final boolean clearLocalTokensOnly, final String browserPackage) {
-        LocalDataManager.clearCache(pool.awsKeyValueStore, context, pool.getAppId(), userId);
         if (!clearLocalTokensOnly) {
+            endSession(browserPackage);
+        }
+
+        // Delete local cache
+        LocalDataManager.clearCache(pool.awsKeyValueStore, context, pool.getAppId(), userId);
+    }
+
+    /**
+     * Ends current browser session.
+     * @param browserPackage browser package to launch sign-out endpoint from.
+     * @throws AuthClientException if sign-out redirect fails to resolve.
+     */
+    private void endSession(final String browserPackage) throws AuthClientException {
+        boolean redirectReceived;
+        try {
+            cookiesCleared = new CountDownLatch(1);
             launchSignOut(pool.getSignOutRedirectUri(), browserPackage);
+            if (!isRedirectActivityDeclared()) {
+                cookiesCleared.countDown();
+            }
+            redirectReceived = cookiesCleared.await(REDIRECT_TIMEOUT_SECONDS, TimeUnit.SECONDS);
+        } catch (InterruptedException e) {
+            throw new AuthNavigationException("User cancelled sign-out.");
+        }
+        if (!redirectReceived) {
+            throw new AuthServiceException("Timed out while waiting for sign-out redirect response.");
```

Comment:
These are runtime exceptions so they do not introduce any breaking change.

Neither exception will ever be thrown if they are not using the new `HostedUIRedirectActivity` either.